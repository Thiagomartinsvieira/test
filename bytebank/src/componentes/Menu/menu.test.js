import { render, screen } from '@testing-library/react'
import Menu from './index'

test('To be render a link to the initial page', () => {
    render(<Menu />)
    const linkInitialPage = screen.getByText('Inicial')
    expect(linkInitialPage).toBeInTheDocument()
})

test('To be render a list of links', () => {
    render(<Menu />)
    const listOfLinks = screen.getAllByRole('link')
    expect(listOfLinks).toHaveLength(4)
})

test("Dont't have to render a link to extract", () => {
    render(<Menu />)
    const linkExtract = screen.queryByTestId('Extract')
    expect(linkExtract).not.toBeInTheDocument()
})

test("To be render a list of links with a class link", () => {
    render(<Menu/>)
    const links = screen.getAllByRole("link")
    links.forEach((link) => expect(link).toHaveClass('links'));
    expect(links).toMatchSnapshot()
})
