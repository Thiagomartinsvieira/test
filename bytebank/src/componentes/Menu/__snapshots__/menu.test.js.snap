// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`To be render a list of links with a class link 1`] = `
Array [
  <a
    class="links"
    href="/"
  >
    Inicial
  </a>,
  <a
    class="links"
    href="/"
  >
    Transferências
  </a>,
  <a
    class="links"
    href="/"
  >
    Investimentos
  </a>,
  <a
    class="links"
    href="/"
  >
    Outros serviços
  </a>,
]
`;
